StRoot/StSpinPool/StFcsEpdQaMaker/StFcsEpdQaMaker.cxx

```diff
@@ -38,26 +38,37 @@ Int_t StFcsEpdQaMaker::Init(){
     return kStFatal;
   }
   
-  if(mFilename[0]==0 && mRun>0){
+  if(mFilename[0]=='0' && mRun>0){
       int yday=mRun/1000;
       sprintf(mFilename,"%d/%d.epdqa.root",yday,mRun);
       printf("StFcsEpdQaMaker::Init - Opening %s\n",mFilename);
   }
   mFile=new TFile(mFilename,"RECREATE");
-
-  char t[100], tt[100];
-  for(int i=0; i<16; i++){
-    sprintf(t,"QtDepAdcCh%d",i);
-    sprintf(tt,"Dep01Ch%d-PP10TT%d; QTADC; DEP Fit Integral",i,i*2);
-    mQtDepA[i] = new TH2F(t,tt,256,0,1024,256,0,1024*4);
-    sprintf(t,"QtDepTacCh%d",i);
-    sprintf(tt,"Dep01Ch%d-PP10TT%d; QTTAC; DEP Peak Timebin",i,i*2);
-    mQtDepT[i] = new TH2F(t,tt,100,0,3000,100,45,56);
-    sprintf(t,"QtDepRatCh%d",i);
-    sprintf(tt,"Dep01Ch%d-PP10TT%d; DEP Peak Timebin; QTADC/DEPIntg;",i,i*2);
-    mQtDepR[i] = new TH2F(t,tt,100,44,57,100,0.0,0.8);
-  }
   
+  char t[100], n[100];
+  char *cNS[2]={"N","S"};
+  for(int det=kFcsPresNorthDetId; det<=kFcsPresSouthDetId; det++){
+      for(int id=0; id<kFcsPresMaxId; id++){
+	  char name[100];
+	  int ns= mFcsDb->northSouth(det);
+	  mFcsDb->getName(det,id,name);	  
+	  sprintf(t,"EPDADC_%1s%03d",cNS[ns],id);
+	  sprintf(n,"%s; QTADC; DEP Fit Integral",name);
+	  mQtDepA[ns][id] = new TH2F(t,n,64,0,1024,64,0,1024*4);
+	  sprintf(t,"EPDTAC_%1s%03d",cNS[ns],id);
+	  sprintf(n,"%s; QTTAC; DEP Fit Peak Timebin",name);
+	  mQtDepT[ns][id] = new TH2F(t,n,50,0,3000,50,45,56);
+          sprintf(t,"EPDRatio_%1s%03d",cNS[ns],id);
+          sprintf(n,"%s; DEP Peak Timebin; QTADC/DEPIntg",name);
+	  mQtDepR[ns][id] = new TH2F(t,n,50,44,57,50,0.0,0.8);
+      }
+  }
+  mQtDepA[0][kFcsPresMaxId] = new TH2F("EPDADCc","EPDADC QTc; QTc ADC; DEP Fit Integral",64,0,1024,64,0,1024*4);
```

Comment:
.. likewise, AddHist( ... ) would place these in the maker's dataset hierarchy ... any reason this is being avoided ?